ui/e2e/README.md

```diff
@@ -0,0 +1,52 @@
+# End-to-end tests
+
+This package contains end-to-end tests for Perses written using [https://playwright.dev/](Playwright).
+
+## Directory structure
+
+- `src`
+  - `fixtures` - [Playwright test fixtures](https://playwright.dev/docs/test-fixtures) live here. These are useful for managing common setup and teardown patterns across many tests. See `pages` for managing common page interactions and selectors.
+  - `pages` - [Page object models](https://playwright.dev/docs/pom) live here. These are classes that wrap selectors, page interactions, and other common patterns associated with a page. This helps reduce code duplication and improve test maintenance. In addition to pages, it also includes classes for large, complex page elements (e.g. panel editor) that benefit from their own wrappers.
+  - `tests` - [Playwright tests](https://playwright.dev/docs/writing-tests) live here and are named following the pattern `testName.spec.ts`.
+- `playwright.config.ts` - [Playwright configuration](https://playwright.dev/docs/test-configuration).
+
+## Running tests
+
+### Locally
+
+The tests depend on the local development servers (backend and UI) to test against.
+
+- Start the backend server from the project root: `./scripts/api_backend_dev.sh`
```

Comment:
based on the playwright configuration, it doesn't seem we need to do this step as it's already handled by the command `npm run e2e` right ?

Reply:
The playwright config only does that when running in CI. I implemented it this way for two reasons:
- When doing development, you may want to keep the dev servers long-running instead of having them spin up and stop every time you run the e2e tests.
- Playwright's cleanup doesn't seem to handle turborepo very well, so it's really easy to end up with a ton of zombie node processes on your local machine. This isn't a problem in CI because it's an ephemeral container and github actions destroys any leftover processes before shutting down.

There's a shorter comment to this effect above that bit of the config.